runtime/moonbase/src/governance/referenda.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019-2022 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # Gov2 config
+//! Includes runtime configs for these substrate pallets:
+//! 1. pallet-conviction-voting
+//! 2. pallet-whitelist
+//! 3. pallet-referenda
+
+use super::*;
+use crate::currency::*;
+use frame_support::traits::{EitherOf, MapSuccess};
+use frame_system::EnsureRootWithSuccess;
+use sp_runtime::traits::Replace;
+
+parameter_types! {
+	pub const VoteLockingPeriod: BlockNumber = 7 * DAYS;
+}
+
+impl pallet_conviction_voting::Config for Runtime {
+	type WeightInfo = pallet_conviction_voting::weights::SubstrateWeight<Runtime>;
+	type Event = Event;
+	type Currency = Balances;
+	type VoteLockingPeriod = VoteLockingPeriod;
+	type MaxVotes = ConstU32<512>;
+	type MaxTurnout = frame_support::traits::TotalIssuanceOf<Balances, Self::AccountId>;
+	type Polls = Referenda;
+}
+
+parameter_types! {
+	pub const AlarmInterval: BlockNumber = 1;
+	pub const SubmissionDeposit: Balance = 100 * UNIT * SUPPLY_FACTOR;
+	pub const UndecidingTimeout: BlockNumber = 28 * DAYS;
+}
+
+parameter_types! {
+	pub const MaxBalance: Balance = Balance::max_value();
+}
+pub type TreasurySpender = EitherOf<EnsureRootWithSuccess<AccountId, MaxBalance>, Spender>;
+
+impl origins::pallet_custom_origins::Config for Runtime {}
+
+// purpose of this pallet is to queue calls to be dispatched as by root for later
+impl pallet_whitelist::Config for Runtime {
+	type WeightInfo = pallet_whitelist::weights::SubstrateWeight<Runtime>;
+	type Event = Event;
+	type Call = Call;
+	type WhitelistOrigin = EitherOf<
+		EnsureRootWithSuccess<Self::AccountId, ConstU16<65535>>,
```

Comment:
It is just the `EnsureRoot` required as a generic for `EitherOf`. I only used this constant because it is what is used in the Kusama Gov2 PR (used 5 times in that PR with EnsureRootWithSuccess i.e. https://github.com/paritytech/polkadot/pull/5205/files#diff-b2299d42ae2d30e47a174007af1e60a2bf92cec42c2628d7e419a05b2ebbe435R71-R72 )

Reply:
The WhitelistOrigin = Either<EnsureRoot, TechCommitee 2 of 3>. The use of the constants and the map reduce are just to make it work. I followed the conventions of that Kusama Gov2 PR for these conventions, but I do not understand them (the constants).